src/celeritas/Types.cc

```diff
@@ -26,12 +26,7 @@ char const* to_cstring(ActionOrder value)
         "post_post",
         "end",
     };
-    static_assert(
-        static_cast<unsigned int>(ActionOrder::size_) * sizeof(char const*)
-            == sizeof(strings),
-        "Enum strings are incorrect");
-
-    return strings[static_cast<unsigned int>(value)];
+    return to_cstring_impl(value);
```

Comment:
You mean putting a bunch of `static const char* [] =` into a header file? 😱

Reply:
Not exactly :)  The best might be a `constexpr` thing.  For example a function (see for example https://belaycpp.com/2021/08/24/best-ways-to-convert-an-enum-to-a-string/) and we could even completely avoid the double typing by using a ["X" macro](https://en.wikipedia.org/wiki/X_Macro) (used extensively in LLVM)

Reply:
There is also (using a sledge hammer to swat a fly): http://aantron.github.io/better-enums/ where the declaration is nice and tidy:
```
BETTER_ENUM(Channel, int, Red = 1, Green, Blue)
```
(just for the record, the header in support of this is ... 'complicated' :) ).

> The good news is that static assert in the new class will tell us at compile time if the enum gains or loses an item, which is a much more likely change than a rename.

True enough.

> Ugh, I'd much prefer to hide the "less important" stringification code at the risk of the values getting out of sync...

The next order issue is whether we will remember 'all' the steps needed to add a new enumeration.

Reply:
> The next order issue is whether we will remember 'all' the steps needed to add a new enumeration.

The compiler error shows up at the same point in the code where the single change has to be made.

Since these are all a very minor set of tradeoffs for what is primarily debugging output, and especially since it's all just a consequence of C++'s lack of reflection/introspection (see the [reflection TS](https://en.cppreference.com/w/cpp/experimental/reflect)) perhaps we should save our energy and wait for simplifying changes until that's part of the language.

Reply:
>  where the single change has to be made.

This PR has reduce the boilerplate significantly but it is still a tad bit more than one line (I meant the `to_cstring` functions) and will to compilation failures only if `to_cstring` is actually used/tested not if it is just missing.

Reply:
> Since these are all a very minor set of tradeoffs for what is primarily debugging output .... wait for simplifying changes until that's part of the language.

That is a very fair point.  I would still make a different choice (a simple const-expr is basically the same complexity has right now and a simplistic macro or template might be sufficient) but it is still a tad bit more work than leaving as is and at any rate could/should (if ever) be done in a different PR.